test/node-http-test.js

```diff
@@ -0,0 +1,285 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: 'off' */
+
+'use strict';
+
+const assert = require('bsert');
+const Network = require('../lib/protocol/network');
+const FullNode = require('../lib/node/fullnode');
+const rules = require('../lib/covenants/rules');
+const common = require('./util/common');
+const network = Network.get('regtest');
+
+const { NodeClient, WalletClient } = require('hs-client');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const wallet = wclient.wallet('primary');
+
+describe('Node http', function() {
+  this.timeout(10000);
+  const witnessedBlocks = {};
+  let NAME0, NAME1;
+  let node, mineBlocks, cbAddress;
+
+  beforeEach(async () => {
+    assert.equal(network.names.auctionStart, 0);
+
+    node = new FullNode({
+      memory: true,
+      apiKey: 'foo',
+      network: 'regtest',
+      workers: true,
+      plugins: [require('../lib/wallet/plugin')]
+    });
+
+    node.on('connect', (entry, block) => {
+      const blockHash = block.hash().toString('hex');
+      witnessedBlocks[blockHash] = blockHash;
```

Comment:
Why not use a `Set` if you are using an object for membership testing? There is also a `BufferSet` in the `buffer-map` library